product_variant_inactive/views/product_variant_view.xml

```diff
@@ -0,0 +1,35 @@
+<odoo>
+   <record model="ir.ui.view" id="view_stock_product_tree">
+        <field name="model">product.product</field>
+        <field name="inherit_id" ref="stock.view_stock_product_tree"/>
+        <field name="arch" type="xml">
+            <!-- Add grey color for inactive record.
+                 Variants are mainly used with stock, 
+                 Then we also embedds color attributes of module stock -->
+            <xpath expr="//tree" position="attributes">
+                <attribute name="colors" translation="off">grey:active==False;red:virtual_available&lt;0;blue:virtual_available&gt;=0;black:virtual_available&gt;=0 </attribute>
+            </xpath>
+        </field>
+    </record>
+
+
+    <record model="ir.ui.view" id="detail_field_custom_in_view">
+        <field name="model">product.product</field>
+        <field name="inherit_id" ref="product.product_product_tree_view"/>
+        <field name="arch" type="xml">
+            <xpath expr="//tree" position="inside">
+                <field name="active" invisible ="1"/>
+                <button type="object" name="button_activate" invisible="1"
+                        attrs="{'invisible': [('active','=',True)]}" icon="fa-ban"/>
+                <button type="object" name="button_deactivate" invisible="1"
```

Comment:
@bealdav As I said, not blocking, just trying to understand the use case. I've forced a rebuild on runbot just to test.